components/UI/actions/socialmediaActions.tsx

```diff
@@ -1,62 +1,58 @@
-import ClickableTwitterIcon from "./clickable/clickableTwitterIcon";
-import { isStarkRootDomain } from "starknetid.js/packages/core/dist/utils";
-import { cairo } from "starknet";
+import React, { FunctionComponent, useEffect, useState } from 'react';
+import ClickableSocialIcon from './clickable/clickableSocialIcon';
+import { isStarkRootDomain } from 'starknetid.js/packages/core/dist/utils';
+import { cairo } from 'starknet';
+
+type SocialPlatform = 'twitter' | 'discord' | 'github';
 
 type SocialMediaActionsProps = {
   identity: Identity;
 };
 
+type SocialProfiles = {
+  twitter?: string;
+  discord?: string;
+  github?: string;
+};
+
 const SocialMediaActions: FunctionComponent<SocialMediaActionsProps> = ({
   identity,
 }) => {
-  const [twitter, setTwitter] = useState<string | undefined>();
-  const [discord, setDiscord] = useState<string | undefined>();
-  const [github, setGithub] = useState<string | undefined>();
+  const [socialProfiles, setSocialProfiles] = useState<SocialProfiles>({
+    twitter: undefined,
+    discord: undefined,
+    github: undefined,
+  });
 
   useEffect(() => {
     if (isStarkRootDomain(identity?.domain.domain)) {
-      identity?.verifier_data?.forEach((verifier) => {
-        if (cairo.felt(verifier.field) === cairo.felt("twitter")) {
-          setTwitter(verifier.data);
-        }
-        if (cairo.felt(verifier.field) === cairo.felt("discord")) {
-          setDiscord(verifier.data);
-        }
-        if (cairo.felt(verifier.field) === cairo.felt("github")) {
-          setGithub(verifier.data);
+      const newProfiles: SocialProfiles = {};
+
+      identity?.verifier_data?.forEach(verifier => {
+        const field = cairo.felt(verifier.field) as SocialPlatform;
+        if (field in newProfiles) {
+          newProfiles[field] = verifier.data;
         }
       });
+
+      setSocialProfiles(newProfiles);
     }
   }, [identity]);
 
   return (
     <div className="flex flex-row justify-evenly items-center gap-3">
-      {twitter ? (
-        <ClickableTwitterIcon
-          width="16"
-          domain={identity?.domain.domain}
-          twitterId={twitter}
-        />
-      ) : null}
-
-      {discord ? (
-        <ClickableDiscordIcon
-          width="16"
-          domain={identity?.domain.domain}
-          discordId={discord}
-        />
-      ) : null}
-
-      {github ? (
-        <ClickableGithubIcon
-          width="16"
-          domain={identity?.domain.domain}
-          githubId={github}
-        />
-      ) : null}
+      {Object.entries(socialProfiles).map(([platform, profileId]) => {
+        const socialPlatform = platform as SocialPlatform;
+        return profileId ? (
+          <ClickableSocialIcon
+            key={platform}
+            platform={socialPlatform}
+            width="16"
+            profileId={profileId as string}
+            domain={identity?.domain.domain}
+          />
+        ) : null;
+      })}
     </div>
   );
 };
```

Comment:
this component doesn't work, both [my](https://app.starknet.id/identities/8) twitter/discord and github are verified but nothing appears on [the build](https://starknet-quest-git-fork-lauchaves-merge-clickabl-c763bb-lfglabs.vercel.app/).

Please verify yours on a valid starknet id and test before sending your PR

Reply:
hey @fricoben fixed some errors there, I tested with Nico's one looks good 

Prod 

![image](https://github.com/user-attachments/assets/1ba79219-8a84-4260-af89-d8f877880699)


Local 

![image](https://github.com/user-attachments/assets/715a85b6-c7b5-4807-b17b-c2c2dadba0cb)